src/main/scala/is/hail/expr/ir/functions/UtilFunctions.scala

```diff
@@ -131,6 +40,31 @@ object UtilFunctions extends RegistryFunctions {
       InsertFields(s, annotations.asInstanceOf[MakeStruct].fields)
     }
 
+    registerCode("toInt32", TBoolean(), TInt32()) { case (_, x: Code[Boolean]) => x.toI }
+    registerCode("toInt64", TBoolean(), TInt64()) { case (_, x: Code[Boolean]) => x.toI.toL }
+    registerCode("toFloat32", TBoolean(), TFloat32()) { case (_, x: Code[Boolean]) => x.toI.toF }
+    registerCode("toFloat64", TBoolean(), TFloat64()) { case (_, x: Code[Boolean]) => x.toI.toD }
+    registerCode("toInt32", TString(), TInt32()) { case (mb, x: Code[Long]) =>
+      val s = asm4s.coerce[String](wrapArg(mb, TString())(x))
+      Code.invokeStatic[java.lang.Integer, String, Int]("parseInt", s)
+    }
+    registerCode("toInt64", TString(), TInt64()) { case (mb, x: Code[Long]) =>
+      val s = asm4s.coerce[String](wrapArg(mb, TString())(x))
+      Code.invokeStatic[java.lang.Long, String, Long]("parseLong", s)
+    }
+    registerCode("toFloat32", TString(), TFloat32()) { case (mb, x: Code[Long]) =>
```

Comment:
Same for the strings to numbers.

Reply:
these are being tested in `test_str_ops`. I added explicit boolean tests.